### PR TITLE
Fix for issue #150. Port vctl --all-tagged from monolithic code

### PR DESCRIPTION
**Pull request by schandrika**
Fix for issue #150 - port vctl --all-tagged from monolithic code (https://github.com/VOLTTRON/volttron/pull/2882/files)

### CODE REVIEW

File: src/volttron/client/commands/control.py [hunk 1]
```diff
@@ -706,34 +709,50 @@ def disable_agent(opts):
...
 def stop_agent(opts):
+    act_on_agent("stop_agent", opts)
+
+
+def restart_agent(opts):
+    stop_agent(opts)
+    start_agent(opts)
+
+
+def act_on_agent(action, opts):
```
Comment (craig8) [inline src/volttron/client/commands/control.py:724]: Please document the new "act_on_agent" function now including what is expected in opts and how this method works.

File: src/volttron/client/commands/control.py [hunk 2]
```diff
@@ -145,13 +145,13 @@ def escape(pattern):
 
 
 def filter_agents(agents, patterns, opts):
```
Comment (craig8) [inline src/volttron/client/commands/control.py:147]: Please document the filter_agents method to have the different options be called out in the docs.

File: src/volttron/client/commands/control.py [hunk 3]
```diff
@@ -706,34 +709,50 @@ def disable_agent(opts):
...
+            return
+
+    if action == "stop_agent":
+        if pid and status is None:
+            _stdout.write(f"Stopping {agent.uuid} {agent.name}\n")
+            call(action, agent.uuid)
+            return
 
 
 def run_agent(opts):
```
Comment (craig8) [inline src/volttron/client/commands/control.py:758]: Please document the run_agent(opts) method

**Comment by craig8** [comment]
without the platform running I think you should be able to get a help from vctl

```
vctl --help
```

not sure if we can separate all of the commands out but the top level should work?  This might be a different issue to be made though.

**Comment by schandrika** [comment]
Created new issue #155

**Pull request merged**